projects/storefrontlib/src/events/product/product-page-event.builder.ts

```diff
@@ -111,8 +105,7 @@ export class ProductPageEventBuilder {
         return searchResults$.pipe(
           map((searchResults) =>
             createFrom(SearchPageResultsEvent, {
-              ...this.createDeprecatedPageEvent(navigationEvent),
-              navigation: { ...navigationEvent },
+              navigation: navigationEvent,
               ...{
                 searchTerm: searchResults?.freeTextSearch,
                 numberOfResults: searchResults?.pagination?.totalResults,
```

Comment:
Do we want to make the change to remove the hard mode warning?

Reply:
I added some default values, please let me know what do you think about it.
But I think this is a drawback of our models, as some properties are not optional in reality, and this can be addressed with a wider refactor where we would fix the problem in the root (in our model interfaces).